packages/core/src/external/cda/remove-b64.ts

```diff
@@ -37,18 +47,45 @@ export function removeBase64PdfEntries(payloadRaw: string): {
         if (comp.section.entry) {
           //eslint-disable-next-line @typescript-eslint/no-explicit-any
           comp.section.entry = toArray(comp.section.entry).filter((entry: any) => {
-            const mediaType = entry.act?.text?.["_mediaType"]?.trim().toLowerCase();
-            if (
-              (BINARY_MIME_TYPES.includes(mediaType) || mediaType == undefined) &&
-              (entry.act?.text?.["_representation"]?.trim().toLowerCase() ===
-                b64Representation.toLowerCase() ||
-                entry.organizer?.component?.observationMedia?.value?.["_representation"]
-                  ?.trim()
-                  .toLowerCase() === b64Representation.toLowerCase())
-            ) {
-              isRemovedEntries = true;
-              b64Attachments.push(entry);
-              return false;
+            if (isConcernActEntry(entry)) {
+              const act = entry.act;
+              if (
+                isBinaryMimeTypeOrUndefined(act.text?._mediaType) &&
+                isB64Representation(act.text?._representation)
+              ) {
+                isRemovedEntries = true;
+                b64Attachments.acts.push(act);
+                b64Attachments.total++;
+                return false;
+              }
+            } else if (isObservationOrganizer(entry)) {
+              const mediaComponents = getMediaObservations(entry.organizer);
```

Comment:
You iterate through all components twice. Once here to get mediaObservations, once at line 67 to get non-mediaObservations. These should be combined and you should iterate just once producing both array.